### PR TITLE
rust/sip: register parser for tcp v6

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3351

Describe changes:
- rebase
- formatting

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1341

### CODE REVIEW

File: rust/src/sip/sip.rs [hunk 1]
```diff
@@ -388,4 +569,22 @@ pub unsafe extern "C" fn rs_sip_register_parser() {
     } else {
         SCLogDebug!("Protocol detecter and parser disabled for SIP/UDP.");
```
Comment (jasonish) [inline rust/src/sip/sip.rs:570]: The only thing I'd change now I think is this line and the one further below.

```
Protocol detection and parsing disabled for UDP SIP
```
and likewise for the TCP.  Or just bring correct spelling to both. Easy to do with in a staging commit though.

**Review by jasonish — APPROVED** [review]
I think this is OK.  I'm not sure if its suitable for 7.0.x as it is much like adding a new parser, but it is a reasonable enhancement to 7.0 as well.  Will need to be discussed.

**Comment by catenacyber** [comment]
Why is the CI red ?

**Comment by catenacyber** [comment]
> Why is the CI red ?

Maybe you need to rebase the S-V PR ?..

File: rust/src/sip/sip.rs [hunk 2]
```diff
@@ -335,6 +475,23 @@ pub unsafe extern "C" fn rs_sip_parse_request(
...
+#[no_mangle]
+pub unsafe extern "C" fn rs_sip_parse_request_tcp(
+    flow: *const core::Flow, state: *mut std::os::raw::c_void, pstate: *mut std::os::raw::c_void,
+    stream_slice: StreamSlice, _data: *const std::os::raw::c_void,
+) -> AppLayerResult {
+    if stream_slice.is_empty() {
+        if AppLayerParserStateIssetFlag(pstate, APP_LAYER_PARSER_EOF_TS) > 0 {
+            return AppLayerResult::ok();
+        } else {
+            return AppLayerResult::err();
```
Comment (catenacyber) [inline rust/src/sip/sip.rs:487]: How can we end up here ?

**Comment by catenacyber** [comment]
Commit `rust/sip: register parser for sip` should reference the redmine ticket number in its message

**Comment by catenacyber** [comment]
I think commit `[eve/schema: add sip_tcp](https://github.com/OISF/suricata/pull/9387/commits/50b5d79a8a933f8ae7c800b241caeac4275d85bf)` should be squashed into `[rust/sip: register parser for sip](https://github.com/OISF/suricata/pull/9387/commits/5e1622835bd9c856fd7eee456f9ba68eb1913361)`

File: suricata.yaml.in [hunk 3]
```diff
@@ -50,6 +50,7 @@ vars:
     GENEVE_PORTS: 6081
     VXLAN_PORTS: 4789
     TEREDO_PORTS: 3544
+    SIP_PORTS: "[5060, 5061]"
```
Comment (catenacyber) [inline suricata.yaml.in:53]: This is meant to be used in signatures, right ?

**Review by catenacyber — CHANGES_REQUESTED** [review]
Thanks again, looking good.

Here is my quick feedback :
CI : can you make it green  ? (rebasing the S-V PR I guess)
Code : looks good
Commits segmentation : I think some commits can be squashed together
Commit messages :Could you put the redmine ticket reference in the main commit ?
Git ID set : looks fine for me
CLA : I do not have access, but you have already contributed
Doc update : Where should we document this change @jufajardini ? upgrade section ?
Redmine ticket : ok
Rustfmt : thanks 🙇 
Tests : Looks noce, will comment on it
Dependencies added: none

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/9483 right ?

**Pull request closed without merging**